client/components/seo/meta-title-editor/mappings.js

```diff
@@ -50,3 +57,13 @@ const tokenToTag = n =>
 
 export const rawToNative = r => removeBlanks( map( split( r, tagPattern ), tagToToken ) );
 export const nativeToRaw = n => join( map( n, tokenToTag ), '' );
+
+export const toApi = compose(
+	partialRight( mapKeys, rearg( snakeCase, 1 ) ), // 1 -> key from ( value, key )
+	partialRight( mapValues, nativeToRaw )          // native objects to raw strings
+);
+
+export const fromApi = compose(
+	partialRight( mapKeys, rearg( camelCase, 1 ) ), // 1 -> key from ( value, key )
+	partialRight( mapValues, rawToNative )          // raw strings to native objects
+);
```

Comment:
Pretty nice functional? programming here. However, to be honest, it's hard to follow if you don't know all the functions used here. I had to read through lodash doc and it took me quite some time to finally understand this. I'm not sure if we are not sacrificing readability/time spent to review and understand this for code beauty/quality.